src/utilities/findBreakingChanges.js

```diff
@@ -169,46 +177,69 @@ export function findArgChanges(
 
       oldTypeFields[fieldName].args.forEach(oldArgDef => {
         const newArgs = newTypeFields[fieldName].args;
-        const newTypeArgIndex = newArgs.findIndex(
+        const newArgDef = newArgs.find(
           arg => arg.name === oldArgDef.name
         );
-        const newArgDef = newArgs[newTypeArgIndex];
-
-        const oldArgTypeName = getNamedType(oldArgDef.type);
-        const newArgTypeName = newArgDef ?
-          getNamedType(newArgDef.type) :
-          null;
-
-        if (!oldArgTypeName) {
-          return;
-        }
 
         // Arg not present
-        if (!newArgTypeName) {
+        if (!newArgDef) {
           breakingChanges.push({
             type: BreakingChangeType.ARG_REMOVED,
             description: `${oldType.name}.${fieldName} arg ` +
               `${oldArgDef.name} was removed`,
           });
-
-        // Arg changed type in a breaking way
-        } else if (
-          oldArgTypeName.name !== newArgTypeName.name
-        ) {
-          breakingChanges.push({
-            type: BreakingChangeType.ARG_CHANGED_KIND,
-            description: `${oldType.name}.${fieldName} arg ` +
-              `${oldArgDef.name} has changed type from ` +
-              `${oldArgDef.type.toString()} to ${newArgDef.type.toString()}`,
-          });
-
-        // Arg default value has changed
-        } else if (oldArgDef.defaultValue !== undefined &&
+        } else {
```

Comment:
The logic before was only comparing the results of `getNamedType` which meant we weren't detecting breaking changes when dealing with type modifiers like NonNull or List. The logic unfortunately becomes more complex, but more correct.